binary_operator.go

```diff
@@ -7,45 +7,60 @@ import (
 type binaryOperatorType int
 
 const (
+	// Comparison Operators
 	BinaryOperatorTypeEq = iota
-	BinaryoperatorTypeGT
-	BinaryoperatorTypeLT
-	BinaryoperatorTypeGTE
-	BinaryoperatorTypeLTE
-	BinaryoperatorTypeExists
-	BinaryoperatorTypeNotExists
+	BinaryOperatorTypeGT
+	BinaryOperatorTypeGTE
+	BinaryOperatorTypeIN
+	BinaryOperatorTypeLT
+	BinaryOperatorTypeLTE
+	BinaryOperatorTypeNE
+	BinaryOperatorTypeNIN
+
+	// Element Operators
+	BinaryOperatorTypeExists
+	BinaryOperatorType
+
+	// Evaluation Operators
+	BinaryOperatorTypeRegex
+
+	// Others Operators
+	BinaryOperatorTypeLike
```

Comment:
Operators not in the default implementation of mongo. Check all that it applies.

Reply:
OK, removed the constants of the operators' types does not default of mongo.